pkg/vsphere/disk/disk_manager.go

```diff
@@ -421,3 +428,48 @@ func (m *Manager) devicePathByURI(op trace.Operation, datastoreURI *object.Datas
 
 	return sysPath, nil
 }
+
+func (m *Manager) InUse(op trace.Operation, config *VirtualDiskConfig) (*vm.VirtualMachine, error) {
```

Comment:
Is it possible that the same disk can be found in several hardware config of several vms? I know when we join a volume to a cvm we perform a  `types.VirtualDeviceConfigSpecOperationAdd` if this populates the `vms[i].Config.Hardware.Device` portion then it is possible that we might find multiple matches for this search. I poked a round for a short bit in the `VirtualHardware` api page [here](https://www.vmware.com/support/developer/vc-sdk/visdk41pubs/ApiReference/vim.vm.VirtualHardware.html). If it is possible Could we possibly return a list instead of a single target? 

I also noticed in the VirtualDevice object( [here](https://www.vmware.com/support/developer/vcdk/visdk41pubs/ApiReference/vim.vm.device.VirtualDevice.html) ) that there is a `connectable` attribute which contains some useful information device connection. 
I was wondering if we could possibly check for this for the case of whether the disk is in use rather than looking through only our containers.